astropy/utils/tests/test_data.py

```diff
@@ -2037,6 +2037,17 @@ def test_zero_remote_timeout(temp_cache, valid_urls):
         download_file(TESTURL, timeout=0)
 
 
+def test_no_allow_internet(temp_cache, valid_urls):
+    u, c = next(valid_urls)
```

Comment:
Wait these are local `file://` URLs they should be fine with no `allow_internet`? Or if not the option needs more explanation.

Reply:
`download_file` is never made to handle "download" of local file, TBH. I thought it did, but when I needed to use it myself, I ended up checking if a file is local first or not before calling this function.

Maybe we're digging into a deeper design problem here. Why would we want to cache a local file? 🤷

Reply:
Many of the tests use file:// URLs, and one might want to add a file:// URL to a configurable `sources` list if one were working on a machine without a Net connection.

Reply:
@aarchiba , I am not sure if I understand your comment at https://github.com/astropy/astropy/pull/10632#discussion_r484526418 . Are you suggesting I add to the test, modify the test, or was it just a general comment that does not require a follow-up action? Hope you can clarify. Thanks!